Elastica/Populator/Strategies/NodePopulator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace OpenOrchestra\Elastica\Populator\Strategies;
+
+use OpenOrchestra\Elastica\Indexor\MultipleDocumentIndexorInterface;
+use OpenOrchestra\Elastica\Populator\ElasticaPopulatorInterface;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\Repository\ReadNodeRepositoryInterface;
+
+/**
+ * Class NodePopulator
+ */
+class NodePopulator implements ElasticaPopulatorInterface
+{
+    protected $multipleIndexor;
+    protected $nodeRepository;
+
+    /**
+     * @param MultipleDocumentIndexorInterface $multipleIndexor
+     * @param ReadNodeRepositoryInterface      $nodeRepository
+     */
+    public function __construct(MultipleDocumentIndexorInterface $multipleIndexor, ReadNodeRepositoryInterface $nodeRepository)
+    {
+        $this->multipleIndexor = $multipleIndexor;
+        $this->nodeRepository = $nodeRepository;
+    }
+
+    /**
+     * Populate nodes
+     */
+    public function populate()
+    {
+        $nodes = $this->nodeRepository->findAllCurrentlyPublishedByType(NodeInterface::TYPE_DEFAULT);
+
+        $this->multipleIndexor->indexMultiple($nodes);
```

Comment:
@alavieille : Don't you think this need to be optimized in some way ? I mean hydrate all published nodes could lead to some serious memory problem on site with many many published node, isn't it ?